api/lib/src/lib.rs

```diff
@@ -515,6 +523,72 @@ pub trait BrokerApi: SignedExtrinsicApi + StorageApi + Sized + Send + Sync + 'st
 		self.simple_submission_with_dry_run(pallet_cf_swapping::Call::deregister_as_broker {})
 			.await
 	}
+
+	async fn encode_btc_smart_contract_call(
```

Comment:
Does this take brokers and affiliate brokers into account too?

Reply:
Not yet. For btc it's not trivial, will be added separately.

Reply:
> Not yet. For btc it's not trivial, will be added separately.

Ok, so to start I will just provide this functionality via BaaS, and if I am correct, BTC -> x swaps initiated this way will start out with 0% commission to the affiliate (and main broker), right?

Reply:
Adding broker fees is a separate issue from what is being added here. By the time release this, we may also have support for broker/affiliate fees.

Having said that, there is no real reason to prefer this method over the other unless for whatever reason you don't want to wait 6secs to open a deposit channel.